### ISSUE TITLE
Action hook disregards output value of filter hook

### ISSUE THREAD

**Issue by danielband** [directus/directus#19473]
### Describe the Bug

[Watch](https://sos-de-fra-1.exo.io/directus/Bildschirmaufnahme%202023-08-19%20um%2009.44.25.mov)

### To Reproduce

Setup two flows:
1) Trigger is a filter hook on update/create
- Return the value of the last operation
- Transform payload and return (does not matter how)
2) Trigger is an action hook on update/create

Create or update the collection. The filter modifies the payload correctly but the action input is still the original payload.

### Directus Version

v10.5.3

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by paescuj** [comment]
From https://docs.directus.io/app/flows/operations.html#transform-payload:
> This operation lets you custom define your own JSON payload for use in subsequent operations. This enables you to take multiple sources of data and "tidy them up" into a single payload.

This means the operation needs to be connected to a subsequent step which actually modifies data, like the [Update Data](https://docs.directus.io/app/flows/operations.html#update-data) operation.

**Comment by danielband** [comment]
> From https://docs.directus.io/app/flows/operations.html#transform-payload:
> 
> > This operation lets you custom define your own JSON payload for use in subsequent operations. This enables you to take multiple sources of data and "tidy them up" into a single payload.
> 
> This means the operation needs to be connected to a subsequent step which actually modifies data, like the [Update Data](https://docs.directus.io/app/flows/operations.html#update-data) operation.

@paescuj  Hmm, not sure if there is a misunderstanding. Its not about the transform payload operation. It's about the action hook who receive the 'wrong' (imo) values in the payload.

E.g:
Given a collection 'Test' {field1: string, field2: string} create a new object {field1: 't1', field2: 't2'}

There is a filter hook on 'Test' that returns the following payload {field1: 't3', field2: 't4'}

If you check the collection you'll see {field1: 't3', field2: 't4'}, as it should be.

BUT what the action hook receives as payload is {field1: 't1', field2: 't2'}. Imo, that does not make much sense, because the action hook is supposed to receive the inserted payload and not the payload before filter was called. I think it used to work 'correctly' once upon a time.

I can reproduce the same behavior with hooks as custom extensions.

**Comment by jdjoshuadavison** [comment]
I agree, this is very odd behaviour - would love to see it changed so that the action flow's payload contains the data as per the result of the filter flow.

**Comment by franklaboris** [comment]
I was using v10.4 until a couple of weeks ago and it was working 'correctly' (the action flow receiving the updated payload)
Now in v10.11 it is no longer the case, as described by OP.
